sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/test/scala/com/azure/cosmos/spark/SimpleFileDiagnosticsSlf4jLogger.scala

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.spark
+
+import com.azure.cosmos.spark.diagnostics.SimpleDiagnosticsSlf4jLogger
+import com.nimbusds.jose.util.StandardCharset
+
+import java.io.{BufferedReader, File, FileOutputStream, FileReader}
+import java.nio.charset.StandardCharsets
+import scala.collection.mutable.ListBuffer
+
+// scalastyle:off multiple.string.literals
+
+private[spark] final class SimpleFileDiagnosticsSlf4jLogger(classType: Class[_])
+  extends SimpleDiagnosticsSlf4jLogger(classType: Class[_]) {
+
+  private val thisLock = new Object()
+  private val fileName = SimpleFileDiagnosticsProvider.getLogFileName(classType)
+  private val file = new File(fileName)
+
+  if (!file.exists()) {
+    file.createNewFile()
+  }
+
+  private val fos = new FileOutputStream(file, true)
+
+  override def logInfo(msg: => String): Unit = {
+    super.logInfo(msg)
+    val line = msg + "|||"
+    writeToFile(line)
+  }
+
+  override def logInfo(msg: => String, throwable: Throwable): Unit = {
+    super.logInfo(msg, throwable)
+    val line = if (Option.apply(throwable).isDefined) {
+      msg + "|||" + throwable.toString.filter(_ >= ' ')
+    } else {
+      msg + "|||"
+    }
+
+    writeToFile(line)
+  }
+
+  private def writeToFile(line: String) {
+    thisLock.synchronized {
+      val fileLock = fos.getChannel.lock()
+      try {
+        fos.write(line.getBytes(StandardCharsets.UTF_8))
+        fos.write(System.getProperty("line.separator").getBytes())
+        fos.flush()
+        fileLock.release()
+      } catch {
+        case t: Throwable => fileLock.release()
+      }
+    }
+  }
+
+  def getMessages(): ListBuffer[(String, Option[String])] = {
+    val result = new ListBuffer[(String, Option[String])]
+    thisLock.synchronized {
+      val reader = new BufferedReader(new FileReader(file))
```

Comment:
if the intention is to only capture the logs, you won't need to write to file. you can just buffer in memory in a singleton static buffer.

With in memory construct you won't need to do much file management, or file synchronization. But this works too. :-)

Reply:
I explained in the PR description - in memory buffer won't work because scala test uses different VMs for master and worker

Reply:
I had implemented it first in memory

Reply:
And then realized that the messages were added in the executor - but in the unit test it wasn't there